### PR TITLE
Unit tests tabs

### PR DESCRIPTION
**Pull request by tumf87**
CRUD Panel tabs trait unit tests.

### CODE REVIEW

File: tests/Unit/CrudPanel/CrudPanelColumnsTest.php [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    public function testSetColumnsDetails()
+    {
+        $this->markTestIncomplete('Not correctly implemented');
+
+        // TODO: refactor crud panel sync method
+    }
+
+    public function testSetColumnOrder()
+    {
+        $this->markTestIncomplete('Not yet implemented');
```
Comment (lloy0076) [inline tests/Unit/CrudPanel/CrudPanelColumnsTest.php]: What's the difference between correctly and not yet here?

(also, the test names are horridly close to each other around here - I had to squit to realise one's plural and the other's not - call me 👴).

File: tests/Unit/Models/Article.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+class Article extends Model
+{
+    use CrudTrait;
+
+    protected $table = 'articles';
+
+    /**
+     * Get the author for the article.
+     */
+    public function user()
```
Comment (lloy0076) [inline tests/Unit/Models/Article.php]: Evil author vs user not that it matters I suppose.

File: tests/config/database/migrations/2017_09_05_000000_create_users_table.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class CreateUsersTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('users', function (Blueprint $table) {
```
Comment (lloy0076) [inline tests/config/database/migrations/2017_09_05_000000_create_users_table.php]: I wonder if we should rely on upstream's migration(s) here (and how).

File: tests/Unit/CrudPanel/CrudPanelColumnsTest.php (hunk 1 above)
Comment (tumf87) [reply tests/Unit/CrudPanel/CrudPanelColumnsTest.php]: Those were methods that didn't to what they were expected to do. The `setColumnOrder` is literally an empty method.

File: tests/config/database/migrations/2017_09_05_000000_create_users_table.php (hunk 3 above)
Comment (tumf87) [reply tests/config/database/migrations/2017_09_05_000000_create_users_table.php]: Yup, database testing isn't the easiest thing in the world to conceptualize and configure.
Fortunately, there are a few resource out there:
- https://phpunit.de/manual/current/en/database.html
- https://github.com/orchestral/testbench
- https://github.com/orchestral/testbench-core/tree/3.5/tests/Databases

**Pull request merged**